### DISCUSSION TITLE
课程3代码问题请教

### DISCUSSION THREAD

**Discussion by lambda7xx** [mlc-ai/mlc-zh#68] [Q&A]
1 最初的numpy代码实现为
![image](https://user-images.githubusercontent.com/35867865/180595324-b1f6346d-4a21-43a2-84de-5b9ff220dcad.png)

2 其TVM实现为
![image](https://user-images.githubusercontent.com/35867865/180595338-ec969ac0-2d25-4d4e-999b-2e40b10802d5.png)

3 然后对其做变换
1)numpy形式为
![image](https://user-images.githubusercontent.com/35867865/180595371-e515b21e-d5ae-4dd6-b07f-488ba6f7eb6f.png)
2)对应的TVM操作是
![image](https://user-images.githubusercontent.com/35867865/180595396-c33870a7-d83d-416e-bbcb-46a9f378a0db.png)
![image](https://user-images.githubusercontent.com/35867865/180595401-52edf74c-a03e-4b19-8838-f45c35dc41c6.png)
![image](https://user-images.githubusercontent.com/35867865/180595419-d1a1a030-2d67-4f68-b7f4-bee8ad3d2161.png)
![image](https://user-images.githubusercontent.com/35867865/180595434-09745eac-ec74-44f1-82ca-876ccca3fee9.png)
按我的理解是先对其做split操作，然后在对其做reorder

TVM实现为
![image](https://user-images.githubusercontent.com/35867865/180595486-bd965570-674d-4986-b8c5-0b1744c0300a.png)

3)问题:为什么对j做了split操作后，需要对其做reorder?

如果不做reorder,也行吧，其numpy实现是
![image](https://user-images.githubusercontent.com/35867865/180595779-28525347-c4e1-49a0-bbb5-16f2f963e735.png)

我感觉1)和3)的numpy实现都是一样的，不需要对其做reorder操作吧

**Comment by Hzfengsy** [comment]
1和3的实现是不一样的，因为循环方式不同导致访问内存的顺序不同。所有变换都不影响程序的正确性，只影响程序运行方式。不同的变换结果会导致程序运行速度不一样，但结果都是一样的
